src/ThreeJsPanel.ts

```diff
@@ -605,6 +614,36 @@ export class ThreeJsPanel {
     return this.renderer.getContext().canvas.height;
   }
 
+  getCameraTransform(): CameraTransform {
+    return {
+      position: this.camera.position.toArray(),
+      rotation: this.camera.rotation.toArray() as [number, number, number],
+      up: this.camera.up.toArray(),
+      target: this.controls.target.toArray(),
+      orthoScales: [this.orthoControlsX.scale, this.orthoControlsY.scale, this.orthoControlsZ.scale],
+    };
+  }
+
+  setCameraTransform(transform: Partial<CameraTransform>) {
+    const currentTransform = this.getCameraTransform();
+    // Fill in any missing properties with current transform
+    const newTransform = { ...currentTransform, ...transform };
+
+    this.camera.up = new Vector3().fromArray(newTransform.up).normalize();
+    this.camera.position.set(...newTransform.position);
+    this.camera.setRotationFromEuler(new Euler().fromArray(newTransform.rotation));
+    this.controls.target = new Vector3().fromArray(newTransform.target);
```

Comment:
1. Any reason `position` is set in-place but `up` and `target` get new vector objects?
2. I believe you can accomplish the same thing you're doing to `position` without the spread (`...`) by using `fromArray`
3. If no on 1, you could set them all this way

Reply:
`camera.position` is a readonly property, so I can't assign it to a new reference. 

Made some changes, see if this looks better!

Reply:
I like it!